src/overlays/actors/ovl_Bg_Icicle/z_bg_icicle.c

```diff
@@ -21,26 +52,199 @@ const ActorInit Bg_Icicle_InitVars = {
+    Vec3f pos;
+    s32 j;
+    s32 i;
+
+    func_800F0568(globalCtx, &this->dyna.actor.world.pos, 30, 0x28CB);
+
+    for (i = 0; i < 2; i++) {
+        for (j = 0; j < 10; j++) {
+            pos.x = this->dyna.actor.world.pos.x + randPlusMinusPoint5Scaled(8.0f);
+            pos.y = this->dyna.actor.world.pos.y + (Rand_ZeroOne() * arg2) + (i * arg2);
+            pos.z = this->dyna.actor.world.pos.z + randPlusMinusPoint5Scaled(8.0f);
+
+            velocity.x = randPlusMinusPoint5Scaled(7.0f);
+            velocity.z = randPlusMinusPoint5Scaled(7.0f);
+            velocity.y = (Rand_ZeroOne() * 4.0f) + 8.0f;
+
+            EffectSsEnIce_Spawn(globalCtx, &pos, (Rand_ZeroOne() * 0.2f) + 0.1f, &velocity, &accel, &primColor,
+                                &envColor, 30);
+        }
+    }
+}
+
+void BgIcicle_DoNothing(BgIcicle* this, GlobalContext* globalCtx) {
+}
+
+void BgIcicle_Wait(BgIcicle* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.xzDistToPlayer < 60.0f) {
+        this->shiverTimer = 10;
+        this->actionFunc = BgIcicle_Shiver;
+    }
+}
+
+void BgIcicle_Shiver(BgIcicle* this, GlobalContext* globalCtx) {
+    s32 randSign;
+    f32 rand;
+
+    if (this->shiverTimer != 0) {
+        this->shiverTimer--;
+    }
+
+    if (!(this->shiverTimer % 4)) {
+        Audio_PlayActorSound2(this, 0x28D4);
+    }
+
+    if (this->shiverTimer == 0) {
+        this->dyna.actor.world.pos.x = this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z;
+
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->actionFunc = BgIcicle_Fall;
+    } else {
+        rand = Rand_ZeroOne();
+        randSign = (Rand_ZeroOne() < 0.5f ? -1 : 1);
+        this->dyna.actor.world.pos.x = (randSign * ((0.5f * rand) + 0.5f)) + this->dyna.actor.home.pos.x;
+        rand = Rand_ZeroOne();
+        randSign = (Rand_ZeroOne() < 0.5f ? -1 : 1);
+        this->dyna.actor.world.pos.z = (randSign * ((0.5f * rand) + 0.5f)) + this->dyna.actor.home.pos.z;
+    }
+}
+
+void BgIcicle_Fall(BgIcicle* this, GlobalContext* globalCtx) {
+    if ((this->collider.base.atFlags & 2) || (this->dyna.actor.bgCheckFlags & 1)) {
+        this->collider.base.atFlags &= ~2;
+        this->dyna.actor.bgCheckFlags &= ~1;
+
+        if (this->dyna.actor.world.pos.y < this->dyna.actor.floorHeight) {
+            this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+        }
+
+        BgIcicle_Break(this, globalCtx, 40.0f);
+
+        if (this->dyna.actor.params == ICICLE_STALACTITE_REGROW) {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + 120.0f;
+            func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+            this->actionFunc = BgIcicle_Regrow;
+        } else {
+            Actor_MarkForDeath(&this->dyna.actor);
+            return;
+        }
```

Comment:
```suggestion
        } else {
            Actor_MarkForDeath(&this->dyna.actor);
        }
```

Reply:
We tend to always put a return after `Actor_Kill`s